src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+
+
+public class Duke {
+    private final Ui ui;
+    private TaskList taskList;
+    private final Storage storage;
+
+
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage();
+        taskList = new TaskList();
+        try {
+            taskList = new TaskList(storage.load(filePath));
+        } catch (DukeException e) {
+            ui.printError(e.getMessage());
+            taskList = new TaskList();
+        }
+
+    }
+
+    public Duke() {
+        this(Storage.SERIALIZATION_PATH);
+    }
+
+
+    public void run() {
+        Ui.initialize();
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        String[] commandAndParameter = Parser.inputParser(input);
```

Comment:
```suggestion
        String[] commandAndParameter = Parser.parseInput(input);
```
Is there a reason why you deviated from the coding standard where 'Names representing methods must be verbs'?